stl/inc/vector

```diff
@@ -2882,6 +2882,62 @@ namespace pmr {
     using vector = _STD vector<_Ty, polymorphic_allocator<_Ty>>;
 } // namespace pmr
 #endif // _HAS_CXX17
+
+#if _HAS_IF_CONSTEXPR
+// VARIABLE TEMPLATE _Is_vb_iterator
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_const_iterator<_Alloc>> = true;
+
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_iterator<_Alloc>> = true;
+
+template <class _FwdIt, class _Ty>
+_CONSTEXPR20 void _Fill_vbool(_FwdIt _First, _FwdIt _Last, const _Ty& _Val) {
+    // Set [_First, _Last) to _Val
+    if (_First == _Last) {
+        return;
+    }
+
+    _Vbase* _UFirst            = const_cast<_Vbase*>(_First._Myptr);
+    const _Vbase* const _ULast = _Last._Myptr;
+
+    const auto _FirstSourceMask = static_cast<_Vbase>(-1) << _First._Myoff;
+    const auto _LastSourceMask  = static_cast<_Vbase>(-1) >> (_VBITS - _Last._Myoff);
+    const auto _FirstDestMask   = static_cast<_Vbase>(~_FirstSourceMask);

```

Comment:
This static_cast is overkill: we know _Vbase isn't going to be 'squirrelly' since that's our type. Maybe we should eliminate that typedef and just say unsigned int or similar?

Reply:
Having dealt with bit manipulations here I always carry a bag full of static casts just to be sure. 

Given that it is at times not "obvious" when we need them I try to err on the safer side

Reply:
IMO more casts is not 'safer' since casts themselves are a potential source of bugs. This isn't an over my dead body issue though.